### PR TITLE
[SYSTEMDS-3051] native blas guide

### PR DESCRIPTION
**Pull request by j143**
Enabling Native BLAS on SystemDS instructions

### CODE REVIEW

**Comment by j143** [comment]
cc @corepointer

File: docs/site/native-backend.md [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+SystemDS implements all the matrix operations in Java. This simplifies deployment especially in
+a distributed environment.
+
+In Some cases (such as Deep Neural Networks), to take advantage of native BLAS instead of SystemDS
+internal Java library for performing single node operations such as matrix multiplication, convolution etc.
+
+By default, SystemDS will first attempt to use Intel MKL (if installed), and then OpenBLAS (if installed).
+If none of the libraries are available, SystemDS falls back to its internal java library.
+
+> Note: Current SystemDS version only supported on **Linux** platform.
```
Comment (corepointer) [inline docs/site/native-backend.md:48]: There is Windows and Linux support on the x86-64 platform for OpenBLAS, MKL and CUDA.

File: docs/site/native-backend.md [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+#### Intel MKL
+
+[Scripts to set environmental variables](https://software.intel.com/content/www/us/en/develop/documentation/onemkl-linux-developer-guide/top/getting-started/setting-environment-variables/scripts-to-set-environment-variables.html)
+
+For example, to set Intel MKL libraries in the Path:
+
+```sh
+# path adjustments to that script according to your installation
+source /opt/intel/bin/compilervars.sh intel64
```
Comment (corepointer) [inline docs/site/native-backend.md:86]: There is an equivalent script for Windows cmd.exe. I'm not on Windows at the moment so I can't give exact directions but there is a bin folder somewhere in your installed Windows MKL and there is a script like that. If you now launch a bash.exe from that Windows command prompt window, you can make use of MKL when starting SystemDS through our run script.

File: docs/site/native-backend.md [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+Java:
+
+```sh
+-Djava.library.path=/path/to/blas-n-other-dependencies
+```
+
+Spark:
+
+```sh
+spark.executorEnv.LD_LIBRARY_PATH=/path/to/blas-n-other-dependencies
```
Comment (corepointer) [inline docs/site/native-backend.md:102]: Did someone actually test that? I don't run Spark jobs frequently so I'm not absolutely sure. If that is working fine, then we can close that ticket that is still floating around somewhere about native blas on Spark iirc.

**Review by corepointer — COMMENTED** [review]
LGTM except the missing Windows support ;-)
And I saw there were SystemDS tests run for this PR. I think we need to refine our CI there. No need to burn cpu cycles for documentation imho.

**Comment by j143** [comment]
I have added other minor comments to my personal TODO list. I merged this one for now. Thanks. 😺

**Pull request merged**